modules/compute/virtual-machine-scale-set/main.bicep

```diff
@@ -503,13 +503,13 @@ resource vmss_logAnalyticsWorkspace 'Microsoft.OperationalInsights/workspaces@20
   scope: resourceGroup(split(monitoringWorkspaceId, '/')[2], split(monitoringWorkspaceId, '/')[4])
 }
 
-module vmss_microsoftMonitoringAgentExtension 'extension/main.bicep' = if (extensionMonitoringAgentConfig.enabled) {
-  name: '${uniqueString(deployment().name, location)}-VMSS-MicrosoftMonitoringAgent'
+module vmss_azureMonitorAgentExtension 'extension/main.bicep' = if (extensionMonitoringAgentConfig.enabled) {
```

Comment:
Do we want to replace MMA implementation with AMA (breaking change) or to keep both (new feature)? Is the MMA agent out of support already or just flagged as legacy/not recommended for new deployments?

While I'd see the benefit of discontinuing support for the legacy MMA agent in favor of the recommended AMA one, I wonder how this would be perceived by our users given the many breaking changes we introduced lately and some not necessarily welcoming them with open arms. 😄

@AlexanderSehr @rahalan @ahmadabdalla @fabmas what's your take on this? 
Would you vouch for discontinuing the MMA extension or for keeping its implementation in parallel to the recommended AMA one?

Reply:
Don't really have an option on that. We could also just wait until 0.11. is released this week and merge it in right after as we need to go crazy afterwards anyways.

Reply:
@fabmas following Alexander proposal, you can try to complete this PR now

Reply:
There may be once Bugfix release before that 😄 Ahmad found some issues, so there should be a 0.11.1 release asap

Reply:
Hi @AlexanderSher, @eriqua we should be now at 0,11,1 release. The module is working properly. Can we merge the changes?

Reply:
I believe so, yes. @eriqua?

Reply:
ping @eriqua :D

Reply:
ping @eriqua

Reply:
Looping in @rahalan, AVM VM and VMSS module owner, to finalize the decision on if keeping the legacy MMA support or just remove it as part of migration to AVM.

Reply:
Resolving after reviewing with @eriqua